graphql_compiler/compiler/blocks.py

```diff
@@ -145,7 +148,9 @@ def validate(self):
                     u"{} -> {}".format(key, value)
                 )
 
-    def visit_and_update_expressions(self, visitor_fn):
+    def visit_and_update_expressions(
+        self, visitor_fn: Callable[[Expression], Expression]
+    ) -> "ConstructResult":
```

Comment:
TIL about forward references!
https://www.python.org/dev/peps/pep-0484/#forward-references
I think I've actually run into this issue before and couldn't figure out how to resolve it 😅

Reply:
https://github.com/python/mypy/issues/731 is also good to read about, since it's not currently implemented but is being actively worked on. I ran into this while working on the interpreter module, and it took me a while to diagnose the problem and find that Github issue.